src/common/remove_all_invalid_leading_chars.rs

```diff
@@ -0,0 +1,15 @@
+pub fn remove_all_invalid_leading_chars(string: &str) -> String {
```

Comment:
Small suggestion, I think "all" excess here, because the name is still clear but shorter
```suggestion
pub fn remove_invalid_leading_chars(string: &str) -> String {
```